prober/icmp.go

```diff
@@ -243,13 +277,47 @@ func ProbeICMP(ctx context.Context, target string, module config.Module, registr
 
 	rb := make([]byte, 65536)
 	deadline, _ := ctx.Deadline()
-	if err := socket.SetReadDeadline(deadline); err != nil {
+	if icmpConn != nil {
+		err = icmpConn.SetReadDeadline(deadline)
+	} else {
+		err = v4RawConn.SetReadDeadline(deadline)
+	}
+	if err != nil {
 		level.Error(logger).Log("msg", "Error setting socket deadline", "err", err)
 		return
 	}
 	level.Info(logger).Log("msg", "Waiting for reply packets")
 	for {
-		n, peer, err := socket.ReadFrom(rb)
+		var n int
+		var peer net.Addr
+		var err error
+		var hopLimit float64
+
+		if dstIPAddr.IP.To4() == nil {
+			var cm *ipv6.ControlMessage
+			n, cm, peer, err = icmpConn.IPv6PacketConn().ReadFrom(rb)
+			if cm != nil {
```

Comment:
How can this be nil?

Don't guess what happens on Windows, test it out.

Reply:
When there is no OOB data, m.NN will be zero here:
https://github.com/golang/net/blob/4acb6c075d100d6ce8757083ec7382423f2a7126/ipv6/payload_cmsg.go#L41

If m.NN was never zero, there should not exist that "if m.NN > 0". I'm not an expert here and maybe ICMP might always returns something as OOB. But I cannot be sure.

Now I discovered that windows crossbuilding with go is trivial. I managed to run it in windows. As expected, SetControlMessage returns err="not implemented on windows/amd64", generating a debug message.

In Windows, cm==nil for all my tests. I'll add a debug message when it cm is null.